delphi/translators/for2py/scripts/mod_index_generator.py

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/python
+
+'''
+Generates a module index file. This file describes each module used in a program run.
+The information about each module is represented as a json dictionary and has the
+following fields:
+
+    name:              <module_name>
+    file:              <file_containing_the_module>
+    module:            <list_of_used_modules>
+    symbol_export:     <list_of_symbols_exported_by_module>
+    subprogram_list:   <procedure_mapping_for_module>
+
+The procedure mapping for each subprogram p defined in module M is a mapping from
+each possible tuple of argument types for p to the function to invoke for that argument
+type tuple.
+'''
+
+import sys
+import xml.etree.ElementTree as ET
+from typing import List, Dict
+import re
+
+class ParseState(object):
+    """This class defines the state of the XML tree parsing
+    at any given root. For any level of the tree, it stores
+    the module under which it resides along with the
+    module metadata."""
+
+    def __init__(self, module=None):
+        self.module = module if module is not None else {}
+        self.args = (
+            [arg["name"] for arg in self.module["args"]]
+            if "args" in self.module
+            else []
+        )
+
+    def copy(self, module=None):
+        return ParseState(
+            self.module if module == None else module
+        )
+
+class moduleGenerator(object):
+    def __init__(self):
+        self.asts = []
+        self.useList = []
+        self.functionList = []
+        self.entryPoint = []
+        self.fileName = str()
+
+
+    def parseTree(self, root, state: ParseState) -> bool:
+
+        if root.tag == "module":
+            return self.process_module(root, state)
+
+        elif root.tag == "use":
+            return self.process_use(root, state)
+
+        elif root.tag == "file":
+            return self.process_file(root, state)
+
+        elif root.tag in ["program", "subroutine"]:
+            return self.process_subroutine_or_program(root, state)
+
+        else:
+            status = True
+            for node in root:
+                status = self.parseTree(node, state)
+            return status
+
+    def process_subroutine_or_program(self, root, state):
+        # Initialize the list of used modules to be an empty list for each module
```

Comment:
I think that the fact that the variable `self.useList` is being initialized to an empty list is obvious from the code; this does not need to be repeated in the comment.  What the comment *should* say is: what is the variable `self.useList` ?